builder/xenserver/common/step_wait_for_ip.go

```diff
@@ -29,47 +29,62 @@ func (self *StepWaitForIP) Run(ctx context.Context, state multistep.StateBag) mu
 		return multistep.ActionHalt
 	}
 
-	var ip string
-	err = InterruptibleWait{
-		Timeout:           self.Timeout,
-		PredicateInterval: 5 * time.Second,
-		Predicate: func() (result bool, err error) {
+	go func(c Connection, ui packer.Ui, config CommonConfig) {
+		state.Put("instance_ssh_address", "")
+		var ip string
 
+		for true {
+			time.Sleep(500 * time.Millisecond)
 			if config.IPGetter == "auto" || config.IPGetter == "http" {
-
 				// Snoop IP from HTTP fetch
 				select {
-				case ip = <-self.Chan:
-					ui.Message(fmt.Sprintf("Got IP '%s' from HTTP request", ip))
-					return true, nil
+				case temp_ip := <-self.Chan:
+					if ip != temp_ip {
+						ip = temp_ip
+						ui.Message(fmt.Sprintf("Got IP '%s' from HTTP request", ip))
+						state.Put("instance_ssh_address", ip)
+					}
 				default:
 				}
-
 			}
 
 			if config.IPGetter == "auto" || config.IPGetter == "tools" {
-
 				// Look for PV IP
 				m, err := c.client.VM.GetGuestMetrics(c.session, instance)
 				if err != nil {
-					return false, err
+					continue
 				}
 				if m != "" {
 					metrics, err := c.client.VMGuestMetrics.GetRecord(c.session, m)
 					if err != nil {
-						return false, err
+						continue
 					}
 					networks := metrics.Networks
-					var ok bool
-					if ip, ok = networks["0/ip"]; ok {
-						if ip != "" {
+					if temp_ip, ok := networks["0/ip"]; ok {
+						if temp_ip != "" && ip != temp_ip {
+							ip = temp_ip
 							ui.Message(fmt.Sprintf("Got IP '%s' from XenServer tools", ip))
-							return true, nil
+							state.Put("instance_ssh_address", ip)
 						}
 					}
 				}
 
 			}
+		}
+	}(*c, ui, config)
+
+	time.Sleep(500 * time.Millisecond)
+
+	var ip string
+
+	err = InterruptibleWait{
+		Timeout:           self.Timeout,
+		PredicateInterval: 5 * time.Second,
+		Predicate: func() (result bool, err error) {
+			var ok bool
+			if ip, ok = state.Get("instance_ssh_address").(string); ok && ip != "" {
+				return true, nil
+			}
```

Comment:
It seems like the `InterruptibleWait` could still return a stale value. I'm wondering if it makes sense to introduce a minimum wait time to prevent the `Predicate` from resolving early. This would allow end users to further delay resolving the predicate if the default runtime isn't enough time for their DHCP to converge. I think we could account for the existing `time.Sleep` call above as the default for this value. 

In addition to that, it looks like the Timeout is hard coded to 300 seconds for the xva builder ([source](https://github.com/ddelnano/packer-plugin-xenserver/blob/81c1ea03eec8c7bd11e87d80e1b7822d3ab46a00/builder/xenserver/xva/builder.go#L164)). We should update to be consistent with the [iso version](https://github.com/ddelnano/packer-plugin-xenserver/blob/81c1ea03eec8c7bd11e87d80e1b7822d3ab46a00/builder/xenserver/iso/builder.go#L276) so the maximum can be tuned as well.

Reply:
Good suggestion! I have added a configurable wait time, named "dhcp_wait" that will be used as an initial wait time before executing the `Predicate`. I have also, in a separate commit, aligned the XVA builder so that it uses the same `self.config.InstallTimeout` value as the ISO builder. However, I could not successfully update "config.hcl2spec.go" by executing `make generate`. I got an error that I have not (yet) been able to understand. If you haven't seen it before, I'll spend some time to figure it out, as I assume that I have to include the changes made by `make generate` in this PR? The error I get is the following:

```bash
mapstructure-to-hcl2: common.[Config]: internal error: package "fmt" without types was imported from "github.com/xenserver/packer-builder-xenserver/builder/xenserver/common"
builder/xenserver/common/config.go:1: running "packer-sdc": exit status 1
make: *** [generate] Error 1
```